source/scalar_conservation/hyperbolic_system.h

```diff
@@ -0,0 +1,901 @@
+        }
+
+        /**
+         * Given a conserved state return a primitive state [rho, u_1, ..., u_d,
+         * p]
+         */
+        primitive_state_type to_primitive_state(const state_type &state) const
+        {
+          return state;
+        }
+
+        /**
+         * Transform the current state according to a  given operator
+         * @p lambda acting on a @a dim dimensional momentum (or velocity)
+         * vector.
+         */
+        template <typename Lambda>
+        state_type apply_galilei_transform(const state_type &state,
+                                           const Lambda & /*lambda*/) const
+        {
+          return state;
+        }
+
+      }; /* HyperbolicSystem::View */
+
+      template <int dim, typename Number>
+      friend class View;
+
+      /**
+       * Return a view on the Hyperbolic System for a given dimension @p
+       * dim and choice of number type @p Number (which can be a scalar
+       * float, or double, as well as a VectorizedArray holding packed
+       * scalars.
+       */
+      template <int dim, typename Number>
+      auto view() const
+      {
+        return View<dim, Number>{*this};
+      }
+    }; /* HyperbolicSystem */
+
+
+    /*
+     * -------------------------------------------------------------------------
+     * Inline definitions
+     * -------------------------------------------------------------------------
+     */
+
+
+    inline HyperbolicSystem::HyperbolicSystem(const std::string &subsection)
+        : ParameterAcceptor(subsection)
+    {
+      flux_ = "burgers";
+      add_parameter("flux",
+                    flux_,
+                    "The scalar flux. Valid names are given by any of the "
+                    "subsections defined below");
+
+      riemann_solver_greedy_wavespeed_ = false;
+      add_parameter(
+          "riemann solver greedy wavespeed",
+          riemann_solver_greedy_wavespeed_,
+          "Use a greedy wavespeed estimate instead of a guaranteed upper bound "
+          "on the maximal wavespeed (for convex fluxes).");
+
+      riemann_solver_averaged_entropy_ = false;
+      add_parameter(
+          "riemann solver averaged entropy",
+          riemann_solver_averaged_entropy_,
+          "In addition to the wavespeed estimate based on the Roe average and "
+          "flux gradients of the left and right state also enforce an entropy "
+          "inequality on the averaged Krŭzkov entropy.");
+
+      riemann_solver_random_entropies_ = 0;
+      add_parameter(
+          "riemann solver random entropies",
+          riemann_solver_random_entropies_,
+          "In addition to the wavespeed estimate based on the Roe average and "
+          "flux gradients of the left and right state also enforce an entropy "
+          "inequality on the prescribed number of random Krŭzkov entropies.");
+
+      riemann_solver_approximation_delta_ = 1.0e-10;
+      add_parameter("riemann solver approximation delta",
```

Comment:
I need to refresh on the scalar conservation law theory, but why does this need to be a user parameter?

Reply:
I will refactor this into an internally chosen parameter.